marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -3957,6 +3957,51 @@ public RESTServiceResultIterator postIteratedResource(RequestLogger reqlog,
       reqlog, path, transaction, params, input, outputMimetypes);
   }
 
+    public RESTServiceResultIterator postMultipartForm(
```

Comment:
This copies a good chunk of `postIteratedResourceImpl`, but I didn't want to factor out the common stuff until I know this is working for all scenarios. The main thing this does is construct a multipart/form-data request.